### PR TITLE
OBPIH-2068 Fix cancel button on add product source pup-up

### PR DESCRIPTION
**Pull request by DominikaHatala**

### CODE REVIEW

File: grails-app/views/productSupplier/_dialog.gsp [hunk 1]
```diff
@@ -168,6 +168,6 @@
             <g:actionSubmit action="save" class="button icon approve" value="Save" id="save">${warehouse.message(code: 'default.button.save.label', default: 'Save')}</g:actionSubmit>
         </g:else>
         &nbsp;
-        <a href="#" class="close-dialog" dialog-id="product-supplier-dialog">${warehouse.message(code: 'default.button.cancel.label', default: 'Cancel')}</a>
+        <button name="cancelDialog" type="reset" onclick="$('.ui-dialog-titlebar-close').click();" class="button"><warehouse:message code="default.button.cancel.label"/></button>
```
Comment (jmiranda) [inline grails-app/views/productSupplier/_dialog.gsp]: Whenever something like this isn't working, it would be better to investigate the reason it's not working rather than come up with a new solution. While this new solution is fine, it leaves the underlying problem in place and we'll eventually have more of these "Fix cancel button on dialog X". 

Since there's already a mechanism for closing the dialog, we just need to fix it so it works for all dialogs. Do a search for "close-dialog" in the source code. You'll find a bunch of buttons with class "close-dialog" and a jquery $(".btn-close-dialog") event handler. So it seems like at some point I was refactoring the close-dialog to be btn-close-dialog in one place but didn't apply the changes everywhere. So find all of the close-dialog buttons and change the class to btn-close-dialog. That alone won't solve the problem because we most of these close-dialog buttons were using data-id to pass the ID of the dialog to the close-dialog function. My recommendation would be to change the target declaration to something like 
```
var target = $(this).data("target") || $(this).data("id") || "#dlgShowDialog";
```
and hopefully everything will magically work. 

You could also change all .btn-close-dialog elements from data-id to data-target. And honestly that's probably the better approach. You just need to change it everywhere.

So in conclusion I would recommend following these guidelines when fixing a regression.
1) Try to debug the existing mechanism instead of creating a new mechanism
2) Apply the updated convention everywhere 
3. If all else fails, refactor and use the apply the new convention everywhere

Comment (DominikaHatala) [reply grails-app/views/productSupplier/_dialog.gsp]: @jmiranda You're right. While working on this one I first tried to manipulate with class name and this button name but couldn't find the right solution for this to work. Changing class to btn-close-dialog helped. I also changed class names from "close-dialog" to "btn-close-dialog" everywhere.
